src/Microsoft.Diagnostics.Monitoring.RestServer/Controllers/DiagController.cs

```diff
@@ -73,6 +73,17 @@ public Task<ActionResult> GetDump(int? pid, [FromQuery]DumpType type = DumpType.
             });
         }
 
+        [HttpGet("gcdump/{pid?}")]
+        public Task<ActionResult> GetGcDump(int? pid, [FromQuery, Range(-1, int.MaxValue)] int timeoutSeconds = 30)
+        {
+            return InvokeService(async () =>
+            {
+                int pidValue = _diagnosticServices.ResolveProcess(pid);
+                Stream result = await _diagnosticServices.GetGcDump(pidValue, TimeSpan.FromSeconds(timeoutSeconds), this.HttpContext.RequestAborted);
+                return File(result, "application/octet-stream", Invariant($"{DateTime.Now:yyyyMMdd\\_HHmmss}_{pidValue}.gcdump"));
```

Comment:
nit: would UtcNow be more appropriate? I realize that it may be confusing, but time zones on servers don't always match the client anyway.